src/libexpr/primops.cc

```diff
@@ -4074,54 +4093,199 @@ void EvalState::createBaseEnv()
 
     /* `builtins' must be first! */
     v.mkAttrs(buildBindings(128).finish());
-    addConstant("builtins", v);
+    addConstant("builtins", v, {
+        .type = nAttrs,
+        .doc = R"(
+          Contains all the [built-in functions](./builtins.md) and values, in order to avoid polluting the global scope.
```

Comment:
```suggestion
          Contains all the [built-in functions](@docroot@/language/builtins.md) and values, in order to avoid polluting the global scope.
```